### PR TITLE
Make the best of page faster

### PR DESCRIPTION
**Pull request by Will-Howard**
The main change here is adding `postsCount` and `postsReadCount` fields to sequences and collections, so we can get these numbers without fetching all the posts. I have also removed some other stuff from the fragments and used `useMulti`s instead of `useSingle`s.

I'm on quite a slow connection so it's hard to judge what the actual speed will be on production, but it's now faster than the frontpage for me, and the limiting factor appears to now be fetching the post bodies which we need for the previews.

The UI should be exactly the same, except some of the number may be off by a small about compared to before. This should just be because draft posts are filtered out in this PR whereas they weren't before, so the new numbers should be correct

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205449867836796) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 1]
```diff
@@ -1416,7 +1417,6 @@ Posts.addView("stickied", (terms: PostsViewTerms, _, context?: ResolverContext)
 Posts.addView("nominatablePostsByVote", (terms: PostsViewTerms, _, context?: ResolverContext) => {
   return {
     selector: {
-      _id: {$in: terms.postIds},
```
Comment (Will-Howard) [inline packages/lesswrong/lib/collections/posts/views.ts:1419]: This is now handled in the default view, so isn't needed here

File: packages/lesswrong/lib/types/collectionTypes.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ interface CollectionBase<
   options: CollectionOptions
   addDefaultView: (view: ViewFunction<N>) => void
   addView: (viewName: string, view: ViewFunction<N>) => void
-  defaultView: ViewFunction<N> //FIXME: This is actually nullable (but should just have a default)
+  defaultView?: ViewFunction<N>
```
Comment (Will-Howard) [inline packages/lesswrong/lib/types/collectionTypes.ts:28]: In `packages/lesswrong/lib/generated/viewTypes.ts` above some collections have `ViewTermsBase` as the view type and some have the named collection view terms. This was previously determined by checking if there were any views defined and falling back to `ViewTermsBase` if not.

This ignored the default view though, and for the `Collections` collection we only need a default view, so I have made the default view explicitly nullable (previously it was already nullable but the type was wrong) and added a check for this in the type generation code

**Review by s-cheng — APPROVED** [review]
Nice! For me it's about the same speed as the home page 👍

File: packages/lesswrong/lib/collections/sequences/schema.ts [hunk 3]
```diff
@@ -169,6 +170,50 @@ const schema: SchemaType<DbSequence> = {
...
+        }
+      );
+
+      return count;
+    }
+  }),
+
+  readPostsCount: resolverOnlyField({
+    type: Number,
+    canRead: ['guests'],
```
[the remark this answers is not in the document]
Comment (Will-Howard) [reply packages/lesswrong/lib/collections/sequences/schema.ts:192]: It still gives a meaningful response for logged out users (0), and it's simpler to not have to handle logged out as a special case so I'm going to leave it as is

File: packages/lesswrong/components/posts/TruncatedAuthorsList.tsx [hunk 4]
```diff
@@ -132,7 +132,7 @@ const recalculate = (
 }
 
 const TruncatedAuthorsList = ({post, expandContainer, classes}: {
```
[the remark this answers is not in the document]
Comment (Will-Howard) [reply packages/lesswrong/components/posts/TruncatedAuthorsList.tsx:134]: Good catch, but this component is going to be killed in the next round of designs so again I'm going to leave it

**Pull request merged**